### ISSUE TITLE
"NotesDataBase.getAllCollections()" broken since version 0.9.30?

### ISSUE THREAD

**Issue by glasen** [klehmann/domino-jna#47]
The above method puts out the following exception since version 0.9.30. Version 0.9.29 works.

```
Initializing Domino JNA with mode Classic
com.mindoo.domino.jna.errors.FormulaCompilationError: . , line=0, column=0, offset=0, length=5, formula=@True
```

**Comment by klehmann** [comment]
Had to change the implementation of getAllCollections() from using the design collection to using an NSF search, because our new methods to create an NSF from scratch do not seem to create the design collection note (it's done when opening the DB in the Notes Client for the first time).

But the new implementation is working for me (Notes 11, macOS Catalina). Which environment are you using?
If you are using the Maven build, could you please try 0.9.31 / 0.9.32? They are already available on Maven Central, since I had to provide small bug fixes / was trying to fix a bug.
And can you please post the full stack trace, at least the part related to com.mindoo.domino.jna.*.

**Comment by glasen** [comment]
I'm running Lotus Domino 11.0.2 on Ubuntu 20.04. Up until version 0.9.29 everything works fine.

Here is a stacktrace:

```
com.mindoo.domino.jna.errors.FormulaCompilationError: . , line=0, column=0, offset=0, length=5, formula=@True
	at com.mindoo.domino.jna.internal.ViewFormulaCompiler.b64_compile(ViewFormulaCompiler.java:92)
	at com.mindoo.domino.jna.NotesSearch.search(NotesSearch.java:467)
	at com.mindoo.domino.jna.NotesSearch.search(NotesSearch.java:172)
	at com.mindoo.domino.jna.NotesDatabase.getAllCollections(NotesDatabase.java:2162)
	at TestNSF.lambda$xmlTest$0(TestNSF.java:67)
	at com.mindoo.domino.jna.gc.NotesGC$3.run(NotesGC.java:557)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.mindoo.domino.jna.gc.NotesGC.runWithAutoGC(NotesGC.java:552)
	at TestNSF.xmlTest(TestNSF.java:59)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:412)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by klehmann** [comment]
That is really weird, just tested the method in Domino 9.0.1 on Linux (CentOS) and Domino 10 on Windows/64 without problems.

I guess

```java
	        FormulaExecution execution = new FormulaExecution("@True");
```

produces the same error? Is there a chance that you can get the value of FormulaCompilationError.m_compileError in the debugger? There's currently no method to read it, I will add that.
Unfortunately, converting the error status to a readable text does not seem to work at all in your setup, which is also very weird.

Which Java version are you using? java.base/ does not look like it's Java 1.8.
If possible, could you try to run the testcase in Java 1.8?

**Comment by klehmann** [comment]
My next tip would be to try this out on a Linux distribution that is supported by Domino (e.g. CentOS), since something seems to be broken with error code translation in your environment. You probably have not compiled a formula before in your code. This probably has never worked.

**Comment by glasen** [comment]
I'm using Java 11 which worked perfectly up until version 0.9.29. And yes, error code translation does not work for me.

`FormulaExecution execution = new FormulaExecution("@True");`

-> Does not work. Same error.

As background:

I'm working as an IT-forensics expert and my Domino-installation is local and is only used to open and process databases which are delivered to my company.  I use a minimized version of Domino (deleted everything which is not absolutely needed) which works very well with your library. 

If the problem lies within my system setup, you can close this bug for now and i will stick to version 0.9.29 which works without any problems.

**Comment by klehmann** [comment]
What you probably can do is take the original code of NotesDatabase.getAllCollections() to read the collections in your own code:
https://github.com/klehmann/domino-jna/blob/0.9.29/domino-jna/src/main/java/com/mindoo/domino/jna/NotesDatabase.java#L1648

You also need to produce your own NotesCollectionSummary class for the collection entries, e.g. by copying this:
https://github.com/klehmann/domino-jna/blob/1c854b009f0d6320a0612a3b6b181621192f45f9/domino-jna/src/main/java/com/mindoo/domino/jna/NotesCollectionSummary.java#L16

**Comment by glasen** [comment]
Hi,

after a year of trying around, i found a very, very simple solution for me problem:

The only thing i needed to do was to create a simple symlink in my Domino-Installation:

```
cd /opt/hcl/domino/notes/latest/linux/res
sudo ln -s de_DE.UTF-8 C
```

That's it. After creating the symlink all methods work as excepted! And i got rid of some warnings too which were printed on the console.

**Comment by klehmann** [comment]
Hi!
Thanks for the info. I actually did the same this week when I set up a CentOS based Domino 12 test server in VMWare Fusion and got hex codes instead of error messages for the exceptions. I compared the Domino 12 environment with our Domino 11 production system and found out that the "C" directory/link was missing.
Unfortunately the error text in my case was not much better than the code (cannot remember, something about a message queue when calling NotesInitExtended) so I decided to use the official Docker image instead, mount my Maven target directory (located in macOS) as a folder and run the test class by hand via a bash script. Worked like a charme.
